test/packaging/windows/msi.tests.ps1

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function Test-Elevated {
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param()
+
+    # if the current Powershell session was called with administrator privileges,
+    # the Administrator Group's well-known SID will show up in the Groups for the current identity.
+    # Note that the SID won't show up unless the process is elevated.
+    return (([Security.Principal.WindowsIdentity]::GetCurrent()).Groups -contains "S-1-5-32-544")
+}
+
+function Invoke-Msiexec {
+    param(
+        [Parameter(ParameterSetName = 'Install', Mandatory)]
+        [Switch]$Install,
+
+        [Parameter(ParameterSetName = 'Uninstall', Mandatory)]
+        [Switch]$Uninstall,
+
+        [Parameter(Mandatory)]
+        [String]$MsiPath,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [HashTable] $Properties
+
+    )
+    $action = "$($PsCmdlet.ParameterSetName)ing"
+    if ($Install.IsPresent) {
+        $switch = '/I'
+    } else {
+        $switch = '/x'
+    }
+
+    $additionalOptions = @()
+    if ($Properties) {
+        foreach ($key in $Properties.Keys) {
+            $additionalOptions += "$key=$($Properties.$key)"
+        }
+    }
+
+    $argumentList = "$switch $MsiPath /quiet /l*vx $script:msiLog $additionalOptions"
+    $msiExecProcess = Start-Process msiexec.exe -Wait -ArgumentList $argumentList -NoNewWindow -PassThru
+    if ($msiExecProcess.ExitCode -ne 0) {
+        $exitCode = $msiExecProcess.ExitCode
+        throw "$action MSI failed and returned error code $exitCode. MSI Log was uploaded as artifact."
+    }
+}
+
+Describe -Name "Windows MSI" -Fixture {
+    BeforeAll {
+        $script:msiX64Path = $env:PsMsiX64Path
```

Comment:
Convert script variable to regular variable as definition in BeforeAll are guaranteed to be present throughout Describe.

Reply:
fixed